torchmultimodal/modules/layers/conv.py

```diff
@@ -0,0 +1,213 @@
+        kernel_size (int or Tuple): size of convolutional filter, same as Conv3d
+        stride (int or Tuple): stride for convolution, same as Conv3d
+        bias (bool): use a bias for convolutional layer or not, same as Conv3d
+
+    Inputs:
+        x (Tensor): input of dims B x C x D1 x D2 x D3
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: _size_3_t,
+        stride: _size_3_t = 1,
+        bias: bool = True,
+        **kwargs
+    ) -> None:
+        super().__init__()
+
+        self.pad_input: Tuple = None
+        self.kernel_size = _triple(kernel_size)
+        self.stride = _triple(stride)
+
+        self.conv = nn.Conv3d(
+            in_channels,
+            out_channels,
+            self.kernel_size,
+            stride=self.stride,
+            bias=bias,
+            **kwargs
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        # Calculate padding needed based on input shape only once to reduce run time
+        if self.pad_input is None:
+            self.pad_input = calculate_same_padding(
+                self.kernel_size, self.stride, x.shape[2:]
+            )
+        return self.conv(F.pad(x, self.pad_input))
+
+
+class SamePadConvTranspose3d(nn.Module):
+    """Performs a same padded transposed convolution on a 3D input. This maintains input shape
+    with unit stride, and multiplies input dims by non-unit stride.
+
+    Code taken from VideoGPT
+    https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
+
+    Args:
+        in_channels (int): number of channels in input, same as Conv3d
+        out_channels (int): number of channels for output, same as Conv3d
+        kernel_size (int or Tuple): size of convolutional filter, same as Conv3d
+        stride (int or Tuple): stride for convolution, same as Conv3d
+        bias (bool): use a bias for convolutional layer or not, same as Conv3d
+
+    Inputs:
+        x (Tensor): input of dims B x C x D1 x D2 x D3
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: _size_3_t,
+        stride: _size_3_t = 1,
+        bias: bool = True,
+        **kwargs
+    ) -> None:
+        super().__init__()
+
+        self.pad_input: Tuple = None
+        self.kernel_size = _triple(kernel_size)
+        self.stride = _triple(stride)
+
+        self.conv = nn.ConvTranspose3d(
+            in_channels, out_channels, kernel_size, stride=stride, bias=bias, **kwargs
+        )
+
+    def forward(self, x) -> Tensor:
+        # Calculate padding needed based on input shape only once to reduce run time
+        if self.pad_input is None:
+            self.pad_input = calculate_same_padding(
+                self.kernel_size, self.stride, x.shape[2:]
+            )
+            self.conv.padding, self.conv.output_padding = calculate_transpose_padding(
```

Comment:
pad_input won't be affected by the padding argument, it should still be none and it will reset the padding as intended agnostic of what the user feeds in. but yeah I agree, let me see if I can make this more explicit